WcaOnRails/app/models/stripe_payment_intent.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class StripePaymentIntent < ApplicationRecord
+  belongs_to :holder, polymorphic: true
+  belongs_to :stripe_transaction
+  belongs_to :user
```

Comment:
Registration `belongs_to` User = the user who initiated the registration
PaymentIntent `belongs_to` User = the user who initiated the payment. These two need not be the same person, because payments can (in theory!) be initiated by a different person, or even be initiated for something that is not a registration at all.

No performance tradeoffs or anything.